test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smppy.py

```diff
@@ -0,0 +1,134 @@
+@pytest.mark.processor("gpu")
+@pytest.mark.integration("smppy")
+@pytest.mark.model("mnist")
+@pytest.mark.skip_cpu
+@pytest.mark.skip_py2_containers
+def test_training_smppy(framework_version, ecr_image, sagemaker_regions):
+    with timeout(minutes=DEFAULT_TIMEOUT):
+        estimator_parameters = {
+            "entry_point": smppy_mnist_script,
+            "role": "arn:aws:iam::920076894685:role/SageMakerRole",
+            "instance_count": 1,
+            "instance_type": INSTANCE_TYPE,
+            "framework_version": framework_version,
+            "hyperparameters": {"epochs": 1},
+            "profiler_config": ProfilerConfig(
+                smprofile_params=SMProfile(cpu_profiling_duration=3600)
+            ),
+            "debug_hook_config": False,
+        }
+        upload_s3_data_args = {"path": training_dir, "key_prefix": "pytorch/mnist"}
+        job_name = utils.unique_name_from_base("test-pt-smppy-training")
+        pytorch, _ = invoke_pytorch_estimator(
+            ecr_image,
+            sagemaker_regions,
+            estimator_parameters,
+            upload_s3_data_args=upload_s3_data_args,
+            job_name=job_name,
+        )
+        _check_and_cleanup_s3_output(pytorch, 40)
+
+
+@pytest.mark.usefixtures("feature_smppy_present")
+@pytest.mark.processor("gpu")
+@pytest.mark.integration("smppy")
+@pytest.mark.model("mnist")
+@pytest.mark.multinode(2)
+@pytest.mark.skip_cpu
+@pytest.mark.skip_py2_containers
+def test_training_smppy_distributed(framework_version, ecr_image, sagemaker_regions):
+    with timeout(minutes=DEFAULT_TIMEOUT):
+        validate_or_skip_pytorchddp(ecr_image)
+        distribution = {"pytorchddp": {"enabled": True}}
+        estimator_parameters = {
+            "entry_point": smppy_mnist_script,
+            "role": "arn:aws:iam::920076894685:role/SageMakerRole",
+            "instance_count": 2,
+            "instance_type": INSTANCE_TYPE,
+            "framework_version": framework_version,
+            "distribution": distribution,
+            "hyperparameters": {"epochs": 1},
+            "profiler_config": ProfilerConfig(
+                smprofile_params=SMProfile(cpu_profiling_duration=3600)
+            ),
+            "debug_hook_config": False,
+        }
+        upload_s3_data_args = {"path": training_dir, "key_prefix": "pytorch/mnist"}
+        job_name = utils.unique_name_from_base("test-pt-smppy-training")
+        pytorch, _ = invoke_pytorch_estimator(
+            ecr_image,
+            sagemaker_regions,
+            estimator_parameters,
+            upload_s3_data_args=upload_s3_data_args,
+            job_name=job_name,
+        )
+        _check_and_cleanup_s3_output(pytorch, 60)
+
+
+def _check_and_cleanup_s3_output(estimator, wait_time):
+    s3 = boto3.client("s3")
+    bucket = estimator.output_path.replace("s3://", "").rstrip("/")
+
+    # Give postprocessing rule some time to complete
+    time.sleep(wait_time)
```

Comment:
This is not robust and could be a source of flaky tests. Why do you need to wait for processing. Are you testing processing job?

Reply:
i check the output of processing (rule-output/DetailedProfilerProcessing-[id]) here. i found that if i did not include the wait time then often i'd get an error indicating the directory was empty, even though when i check in s3 the files are there.